File: app/src/main/java/com/soen490chrysalis/papilio/view/QuizPart1.java
package com.soen490chrysalis.papilio.view;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.soen490chrysalis.papilio.R;

public class QuizPart1 extends AppCompatActivity {

    Button next;
    Button choice1;
    Button choice2;
    TextView question;
    TextView title;
    TextView skip;
    boolean firstClick;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz_part1);

        next = (Button) findViewById(R.id.next);
        choice1 = (Button) findViewById(R.id.choise1);
        choice2 = (Button) findViewById(R.id.choise2);
        question = (TextView) findViewById(R.id.question);
        title = (TextView) findViewById(R.id.textView4);
        skip = (TextView) findViewById(R.id.skip);

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(QuizPart1.this, QuizPart2Activity.class));
            }
        });

        skip.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(QuizPart1.this, MainActivity.class));
            }
        });

        var actionBar = getSupportActionBar();

        if(actionBar != null){
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle("Activity Quiz");
        }



    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        int id = item.getItemId();

        if(id == android.R.id.home){
            this.finish();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/soen490chrysalis/papilio/view/QuizPart1.kt
package com.soen490chrysalis.papilio.view

import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import android.os.Bundle
import android.content.Intent
import android.view.MenuItem
import android.widget.Button
import com.soen490chrysalis.papilio.R
import com.soen490chrysalis.papilio.databinding.ActivityQuizPart1Binding

class QuizPart1 : AppCompatActivity()
{
    private lateinit var binding : ActivityQuizPart1Binding
    private lateinit var next : Button
    private lateinit var outdoorButton : Button
    private lateinit var indoorButton : Button
    private lateinit var question : TextView

    override fun onCreate(savedInstanceState : Bundle?)
    {
        super.onCreate(savedInstanceState)
        binding = ActivityQuizPart1Binding.inflate(layoutInflater)
        setContentView(binding.root)

        next = binding.next
        outdoorButton = binding.choice1
        indoorButton = binding.choice2
        question = binding.question

        var outdoorButtonSelected = false
        var indoorButtonSelected = false

        next.setOnClickListener {
            val intent = Intent(this@QuizPart1, QuizPart2Activity::class.java)
            intent.putExtra("indoorSelected", indoorButtonSelected)
            intent.putExtra("outdoorSelected", outdoorButtonSelected)
            startActivity(intent)
        }

        outdoorButton.setOnClickListener {
            if (outdoorButtonSelected)
            {
                outdoorButton.setCompoundDrawables(null, null, null, null)
            }
            else
            {
                val img = resources.getDrawable(R.drawable.ic_checkmark)
                img.setBounds(0, 0, 60, 60)
                outdoorButton.setCompoundDrawables(null, null, img, null)
            }
            outdoorButtonSelected = !outdoorButtonSelected
        }

        indoorButton.setOnClickListener {
            if (indoorButtonSelected)
            {
                indoorButton.setCompoundDrawables(null, null, null, null)
            }
            else
            {
                val img = resources.getDrawable(R.drawable.ic_checkmark)
                img.setBounds(0, 0, 60, 60)
                indoorButton.setCompoundDrawables(null, null, img, null)
            }
            indoorButtonSelected = !indoorButtonSelected
        }

        val actionBar = supportActionBar
        if (actionBar != null)
        {
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.title = "Activity Quiz"
        }
    }

    override fun onOptionsItemSelected(item : MenuItem) : Boolean
    {
        val id = item.itemId
        if (id == android.R.id.home)
        {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }
}